larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.cc

```diff
@@ -0,0 +1,99 @@
+/**
+ *  @file   larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.cc
+ *
+ *  @brief  Implementation file for the cheated threeD reclustering algorithm.
+ *
+ *  $Log: $
+ */
+
+#include "Pandora/AlgorithmHeaders.h"
+#include "larpandoracontent/LArHelpers/LArMCParticleHelper.h"
+
+#include "larpandoracontent/LArReclustering/CheatedThreeDClusteringTool.h"
+
+
+using namespace pandora;
+
+namespace lar_content
+{
+
+CheatedThreeDClusteringTool::CheatedThreeDClusteringTool()
+{
+}
+
+//------------------------------------------------------------------------------------------------------------------------------------------
+
+bool CheatedThreeDClusteringTool::Run(const Algorithm *const pAlgorithm, std::vector<pandora::CaloHitList*> &newCaloHitListsVector)
+{
+    if (newCaloHitListsVector.empty() || newCaloHitListsVector.size() != 1)
+        return false;
+
+    CaloHitList *initialCaloHitList = newCaloHitListsVector.at(0);
+    newCaloHitListsVector.clear();
+
+    std::map<int, CaloHitList*> McIdCaloHitListMap;
+
+    for (const CaloHit *const pCaloHit : *initialCaloHitList)
+    {
+        const CaloHit *const pParentCaloHit = static_cast<const CaloHit *>(pCaloHit->GetParentAddress());
+
+        int mainMcParticleIndex = this->GetMainMcParticleIndex(pAlgorithm, pParentCaloHit);
+
+        if (McIdCaloHitListMap.find(mainMcParticleIndex) != McIdCaloHitListMap.end()) 
```

Comment:
Could you just use `McIdCaloHitListMap.count(mainMcParticleIndex` to simplify the code here (and I guess make it slightly faster)

Reply:
Yes!